backend/migrations/versions/a991ac7a85a2_drop_remove_group_permissions.py

```diff
@@ -0,0 +1,63 @@
+"""drop_remove_group_permissions
+
+Revision ID: a991ac7a85a2
+Revises: c6d01930179d
+Create Date: 2024-05-02 15:23:01.516657
+
+"""
+
+from sqlalchemy import and_
+from dataall.core.permissions.services.environment_permissions import REMOVE_ENVIRONMENT_GROUP
+from dataall.core.permissions.db.resource_policy.resource_policy_models import ResourcePolicy
+from dataall.core.permissions.services.resource_policy_service import ResourcePolicyService
+from alembic import op
+from sqlalchemy import orm
+from dataall.core.environment.db.environment_models import Environment
+
+# revision identifiers, used by Alembic.
+revision = 'a991ac7a85a2'
+down_revision = 'c6d01930179d'
+branch_labels = None
+depends_on = None
+
+
+def get_session():
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+    return session
+
+
+def upgrade():
+    session = get_session()
+    environments = session.query(Environment).all()
+    for env in environments:
+        admin_group = env.SamlGroupName
+        suspicious_permissions_principals = (
+            session.query(ResourcePolicy.principalId)
+            .filter(
+                and_(
+                    ResourcePolicy.principalType == 'GROUP',
+                    ResourcePolicy.principalId != admin_group,
+                    ResourcePolicy.resourceType == Environment.__name__,
+                    ResourcePolicy.resourceUri == env.environmentUri,
+                )
+            )
+            .all()
+        )
+        for group in suspicious_permissions_principals:
+            permissions = ResourcePolicyService.get_resource_policy_permissions(session, group, env.environmentUri)
+            permissions.remove(REMOVE_ENVIRONMENT_GROUP)
```

Comment:
can we add logic to remove only if `REMOVE_ENVIRONMENT_GROUP` is in the list of permissions

Otherwise we may get an Error when trying to remove this permission for a env group where it does not exist (i.e. they never were assigned `INVITE_ENVIRONMENT_GROUP` beforehand

Reply:
UPDATE: After some testing I believe need to change the way we are removing `REMOVE_ENVIRONMENT_GROUP`, the below worked (or similar logic):

```
permissions = [permission.name for permission in permissions if permission.name != REMOVE_ENVIRONMENT_GROUP]
```

Reply:
I always forget, that in JS remove is protected, but in python it's not! Argh)